app/models/user.rb

```diff
@@ -1,6 +1,27 @@
 class User < ApplicationRecord
+  before_save { email.downcase! }
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+  has_many :posts
+  has_many :likes
+  has_many :comments
+
+  has_many :recieved_requests, foreign_key: "requested_id", 
```

Comment:
Haha I wanna ask you guys a favor, can we correct the spelling of `recieved`? It's supposed to be `received`.

I have a suggestion too, but implementing this is entirely up to you and depending of course on your implementation of friend requests later on in the controller and views. For friendships I can do something like `user1.friends` right? How about if I wanna know how many people `user1` has sent a request to? Or how many requesters `user1` has? I mean right now obviously we can use `sent_requests` and `received_requests`, but for these two methods we still need to type either `requester_id` or `requested_id` in a conditional to get what we want. Probably a method or an additional `has_many` association for requested and requester is possible? Similar to the friendships and friends association you already have.